sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/source_metadata.py

```diff
@@ -4,21 +4,50 @@
 
 # pylint: disable=redefined-builtin,disable=unused-argument
 
-from .timestamp_column_metadata import TimestampColumnMetadata
+from typing import Dict
+
+from azure.ai.ml.entities._feature_set.source_process_code_metadata import SourceProcessCodeMetadata
+from azure.ai.ml.exceptions import ErrorCategory, ErrorTarget, ValidationErrorType, ValidationException
+
 from .delay_metadata import DelayMetadata
+from .timestamp_column_metadata import TimestampColumnMetadata
 
 
 class SourceMetadata(object):
     def __init__(
         self,
         *,
         type: str,
-        path: str,
         timestamp_column: TimestampColumnMetadata,
+        path: str = None,
         source_delay: DelayMetadata = None,
-        **kwargs
+        source_process_code: SourceProcessCodeMetadata = None,
+        dict: Dict = None,
+        **kwargs,
     ):
+        if type != "custom":
+            if not (path and not dict and not source_process_code):
+                msg = f"Only path should be provided when source type is {type}"
```

Comment:
This message is accurate. User might also provide source_delay and time stamp column.

cannot provide source_process_code or kwargs for custom feature source.

Reply:
updated